ibis/pandas/execution/window.py

```diff
@@ -30,12 +30,19 @@
 from ibis.pandas.execution import util
 
 
-def _post_process_empty(scalar, parent, order_by, group_by):
+def _post_process_empty(result, parent, order_by, group_by):
     assert not order_by and not group_by
-    index = parent.index
-    result = pd.Series([scalar]).repeat(len(index))
-    result.index = index
-    return result
+    if isinstance(result, pd.Series):
+        # `result` is a Series e.g. when an analytic operation is being
```

Comment:
can you elabort *when* this happens (e.g. what casuses this result to be a Series vs a scalar). Is this better handled in a dispatch rule though? e.g. to differentiate these cases.

Reply:
I'll make the comments a little clearer

I think if all the `_post_process_xxxx` functions could be made into a dispatch rule that would be pretty nice, but some of them exactly share the same _types_ of input parameters

In terms of a dispatch rule for just `_post_process_empty`, I don't think it'd be any clearer because _I think_ the dispatch rule for the case that result is a scalar would have to be `@_post_process_empty.register(Any)`, which doesn't clearly indicate that that is the scalar case

Reply:
OK this ^ has been done